active_storage/server/models.py

```diff
@@ -1,28 +1,59 @@
+
+from enum import Enum
+from typing import Optional, List, Callable
+from dataclasses import dataclass
+
+from pydantic import BaseModel, Field
 import numpy as np
 
 
-class NumericDataType(np.number):
-    """
-    Model for validating a numpy datatype given as a string.
-    """
-    DATATYPES = {
-        "int32": np.int32,
-        "int64": np.int64,
-        "uint32": np.uint32,
-        "uint64": np.uint64,
-        "float32": np.float32,
-        "float64": np.float64,
-    }
-
-    @classmethod
-    def __get_validators__(cls):
-        yield cls.validate
-
-    @classmethod
-    def validate(cls, v):
-        if not isinstance(v, str):
-            raise TypeError("must be a string")
-        try:
-            return cls.DATATYPES[v]
-        except KeyError:
-            raise ValueError(f"unsupported dtype '{v}'")
+#Use enum which also subclasses string type so that auto-generated OpenAPI schema can determine allowed dtypes
+class AllowedDatatypes(str, Enum):
+    """ Data types supported by active storage proxy """
+    int64 = 'int64'
+    int32 = 'int32'
+    float64 = 'float64'
+    float32 = 'float32'
+    uint64 = 'uint64'
+    uint32 = 'uint32'
+
+    def n_bytes(self):
+        """ Returns the number of bytes in the data type """
+        return np.dtype(self.name).itemsize
+
+
+class RequestData(BaseModel):
+    source: str
+    bucket: str
+    object: str
+    dtype: AllowedDatatypes
+    offset: Optional[int] = 0
```

Comment:
Pydantic is capable of constraining integers using gt, lt, ge, le where appropriate. We should do that.